zc_common/jwt_auth/permissions.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework.permissions import BasePermission
+from .utils import USER_ROLE, STAFF_ROLE, SERVICE_ROLE, ANONYMOUS_ROLE
+
+
+class IsUser(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        return request.user and request.user.is_authenticated() and USER_ROLE in user.roles
+
+
+class IsStaff(IsUser):
+    def has_permission(self, request, view):
+        user = request.user
+        return super(IsStaff, self).has_permission(request, view) and STAFF_ROLE in user.roles
+
+
+class IsOwner(IsUser):
+    def has_object_permission(self, request, view, obj):
+        if not obj.user:
+            return False
+        return str(request.user.id) == str(obj.user)
+
+
+class IsAnonymous(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        return request.user and request.user.is_authenticated() and ANONYMOUS_ROLE in user.roles
+
+
+class IsService(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        return request.user and request.user.is_authenticated() and SERVICE_ROLE in user.roles
+
+
+class IsStaffOrService(IsStaff, IsService):
+    def has_permission(self, request, view):
+        return IsStaff.has_permission(self, request, view) or IsService.has_permission(self, request, view)
```

Comment:
How about this instead:

```python
class BaseAuthPermission(BasePermission):
  def has_permission(self, request, view):
    return request.user and request.user.is_authenticated()


class RoleCheck(BaseAuthPermission):
  passing_roles = [USER_ROLE]

  def has_permission(self, request, view):
    is_authed = super(IsStaff, self).has_permission(request, view)
    has_role = any(role in request.user.roles for role in self.passing_roles)
    return is_authed and has_role


class IsStaff(RoleCheck):
  passing_roles = [STAFF_ROLE]


class IsStaffOrService(RoleCheck):
  passing_roles = [STAFF_ROLE, SERVICE_ROLE]
```

This provides you with a framework for any combination of `OR` conditions for permissions.
If you wanted an `AND`, you'd just add multiple permissions to the relevant view.
`permission_classes = (IsStaff, IsOwner)` etc